docs/layers/path-layer.md

```diff
@@ -102,7 +102,18 @@ Only effective if `getDashArray` is specified. If `true`, adjust gaps for the da
 
 Returns the specified path for the object.
 
-A path is an array of coordinates.
+A path can be one of the following formats:
+
+* An array of coordinates following the GeoJSON [LineString](https://tools.ietf.org/html/rfc7946#section-3.1.4) specification.
+* A flat array or [TypedArray](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray) of numbers, in the shape of `[x0, y0, z0, x1, y1, z1, ...]`. By default, each coordinate is assumed to contain 3 numbers. Attach a `stride` field to the array if it should be sliced differently:
```

Comment:
> Attach a `stride` field to the array if it should be sliced differently:

attaching non-indexed fields to arrays is a little unusual. I did something similar in loaders.gl and it works but doesn't feel like a great API design. 

Also some slight worry about perf as some JS implementations might change the underlying type of the array when no longer contiguously indexed, although I think this may be dated information.

But is there a reasonable case for having different sizes per path (assuming all the incoming data for a layer will be of the same structure)? If not, could this be handle with some layer level accessor descriptor as suggested above?

Reply:
I'm not happy with this either.

I would like to support the 2d points use case though. When using iterable data, we handle missing z with all `getPosition` accessors. Most geospatial data you'll get from GIS or open datasets are 2d, and the users should not have to pad the points with zeros.

Reply:
Added a `is3D` prop to the layer.

Reply:
> and the users should not have to pad the points with zeros.

100% agree.


> Added a is3D prop to the layer.

Great. As for naming, since that name is generic but this covers a very specialized use case, I think this name risks being interpreted as an `extruded` prop by new users. Since it only affects interpretation of `data`, maybe `dataIs3D` is more appropriate?

Probably good enough for now, but we might need to specify per prop later (colors could be a single value in a grey scale map for instance).

Reply:
Good point, it will be even more confusing if the layer has an `extruded` prop.

My other proposal is to add these props to the base Layer class:

```js
pointFormat: 'XYZ',  // alt option `XY`
colorFormat: 'RGBA'  // alt option `RGB`
```

Per https://github.com/uber/luma.gl/pull/850, we can potentially use these settings to reduce attribute size in all layers.

Reply:
> My other proposal is to add these props

Not a bad idea. 

Thoughts on the prop names:
* Perhaps it seems a little arbitrary to use individual props for each of these (i.e. could the list grow?)
* They are only relevant for the binary case, which I consider quite "advanced", but as top-level props could easily become "red herrings" for new users.

The other thing that goes through my mind (as a GLSL coder) is the choice of the `XYZ` and `RGBA` symbols. GLSL supports swizzling which allows us to e.g. write `vec4.zyx` and `color.bgr` to use elements in "swizzled" / reverse order. 

Since these props are clearly closely related to the binary GPU data, so when I see this syntax in a WebGL based framework, I would immediately suspect that we are supporting this, and probably make some assumptions about how this system works that are not true. 

Supporting swizzling is actually a possible extension, so I guess we don't close that door if we go with these string. Alternatively, we could just use numbers for now.

How about collecting this in a separate `dataFormat` prop

```
dataFormat: {
  positions: 2
  colors: 3
}
```

it could even be called `binaryDataFormat` or `dataBinaryFormat` to make sure that new users don't get confused.

We could also document that this prop only influences interpretation, but doesn't trigger changes. This prop would then have minimal perf impact on the layer diffing cycle.